WordPress/gutenberg
# Create Block: Add support for more plugin header fields

## Description


## Testing Instructions





Tested creating block with three exisitng templates in the interactive and non-interactive mode:

```bash
npx wp-create-block --no-wp-scripts
npx wp-create-block test-block --no-wp-scripts
npx wp-create-block -t es5
npx wp-create-block -t es5 test-block
npx wp-create-block -t @wordpress/create-block-tutorial-template --no-wp-scripts
npx wp-create-block -t @wordpress/create-block-tutorial-template test-block --no-wp-scripts
```

Ensure that correct values for `Plugin URI`, `Domain Path` and `Update URI` are included in `index.php` file when provided from interactive prompts. Otherwise those fields should be missing in the plugin header.

Ensure that corrected value for `homepage` is included in the `package.json` file when `Plugin URI` provided from interactive prompts. Otherwise this field should be missing in the `package.json` file.

## Screenshots 

Header fields provided:

<img width="955" alt="Screenshot 2022-02-25 at 17 23 05" src="https://user-images.githubusercontent.com/699132/155750737-bab8fded-2c62-4775-9e10-ea5ac63ff11a.png">

<img width="831" alt="Screenshot 2022-02-25 at 17 23 53" src="https://user-images.githubusercontent.com/699132/155750746-8f033b24-95b7-4a84-a661-a5f651882366.png">


No header fields provided:

<img width="931" alt="Screenshot 2022-02-25 at 17 24 31" src="https://user-images.githubusercontent.com/699132/155750700-b1371eb4-de5f-4132-8173-b0f86ab1569b.png">

## Types of changes


New feature (non-breaking change which adds functionality).


## Checklist:
